lib/plug/conn.ex

```diff
@@ -460,12 +460,15 @@ defmodule Plug.Conn do
   otherwise `{:error, reason}`.
   """
   @spec chunk(t, body) :: {:ok, t} | {:error, term} | no_return
-  def chunk(%Conn{state: :chunked} = conn, ""), do: {:ok, conn}
   def chunk(%Conn{adapter: {adapter, payload}, state: :chunked} = conn, chunk) do
-    case adapter.chunk(payload, chunk) do
-      :ok                  -> {:ok, conn}
-      {:ok, body, payload} -> {:ok, %{conn | resp_body: body, adapter: {adapter, payload}}}
-      {:error, _} = error  -> error
+    case IO.iodata_length(chunk) do
```

Comment:
Thanks for the detail!  I think I agree that the perf of `iodata_length` isn't slow enough to warrant us implementing `empty_iodata?` here (and perhaps getting it wrong; for example, I'm not quite sure if that handles improper lists correctly...).  However, given that the short-circuit version is orders of magnitude faster than `iodata_length` for large iodatas, it seems like Elixir's `IO` module could benefit from an `empty_iodata?/1` function.  If/when that is added to Elixir, plug could be updated in the future to use it.